src/protocol/sort/reshare.rs

```diff
@@ -128,4 +136,43 @@ mod tests {
         }
         assert!(new_reshares_atleast_once);
     }
+
+    #[tokio::test]
+    pub async fn test_reshare_malicious() {
+        let mut rand = StepRng::new(100, 1);
+        let mut rng = rand::thread_rng();
+        let mut new_reshares_atleast_once = false;
+        let world: TestWorld = make_world(QueryId);
+        let [ctx0, ctx1, ctx2] = make_malicious_contexts::<Fp31>(&world);
+        let r = rng.gen::<Fp31>();
+
+        for _ in 0..10 {
+            let secret = rng.gen::<u128>();
+
+            let input = Fp31::from(secret);
+            let shares = share_malicious(input, r, &mut rand);
+            let record_id = RecordId::from(0);
+
+            let [share0, share1, share2] = shares.clone();
+
+            let h0_future = reshare_malicious(ctx0.ctx.clone(), &share0, record_id, Role::H2);
+            let h1_future = reshare_malicious(ctx1.ctx.clone(), &share1, record_id, Role::H2);
+            let h2_future = reshare_malicious(ctx2.ctx.clone(), &share2, record_id, Role::H2);
+
+            let f = try_join_all([h0_future, h1_future, h2_future])
+                .await
+                .unwrap();
+            let output_share = validate_and_reconstruct(f[0].x(), f[1].x(), f[2].x());
+            let output_macs = validate_and_reconstruct(f[0].rx(), f[1].rx(), f[2].rx());
```

Comment:
I think that we're going to need a fixture helper to manage this validation process.